File: (7 kyu) Shortest Word/(7 kyu) Shortest Word.rs
// #1
// fn find_short(s: &str) -> u32 {
//   let words = s.split(" ").collect::<Vec<_>>();
//   let mut min = <usize>::max_value();
//   for w in words {
//       if w.len() < min {
//           min = w.len()
//       }
//   }
//   min as u32
// }
// #2
// fn find_short(s: &str) -> u32 {
//   s
//     .split_whitespace()
//     .min_by_key(|s| s.len())
//     .unwrap()
//     .len() as u32
// }
// #3
fn find_short(s: &str) -> usize {
    s.split_whitespace().map(str::len).min().unwrap()
}

File: (7 kyu) Shortest Word/(7 kyu) Shortest Word.cpp
#include <iostream>
#include <string>
#include <vector>
#include <sstream>

using namespace std;

int find_short(std::string str)
{
    std::vector<std::string> array;
    std::stringstream ss(str);
    std::string x;
    while (ss >> x)
    {
        array.push_back(x);
    }
    int length = array.size();
    int min = 1000;
    for (int i = 0; i < length; i++)
    {
        if (array[i].length() < min)
        {
            min = array[i].length();
        }
    }
    return min;
}